mne/decoding/csp.py

```diff
@@ -111,11 +128,14 @@ def __init__(
         transform_into="average_power",
         norm_trace=False,
         cov_method_params=None,
+        restr_type="restricting",
```

Comment:
We should add these in reasonable places
```suggestion
        *,
        restr_type="restricting",
```
They are good for readability (forcing people to be a bit more verbose with their code), but also required for backward compatibility since now `rank` shows up later in the list of args. Can you make a similar change to all modified classes?